File: Project_24/LifeTracker/backend/src/main/java/com/lifetracker/modules/nutrition/dto/UpdateNutritionGoalsRequest.java
package com.lifetracker.modules.nutrition.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;

import java.math.BigDecimal;

public record UpdateNutritionGoalsRequest(
        @NotNull @DecimalMin("1") BigDecimal calorieGoal,
        @NotNull @DecimalMin("0") BigDecimal proteinGoal,
        @NotNull @DecimalMin("0") BigDecimal carbsGoal,
        @NotNull @DecimalMin("0") BigDecimal fatGoal,
        @NotNull @DecimalMin("0") BigDecimal fiberGoal
) {
}

File: Project_24/LifeTracker/frontend/lib/models/update_nutrition_goals_request.dart
class UpdateNutritionGoalsRequest {
  const UpdateNutritionGoalsRequest({
    required this.calorieGoal,
    required this.proteinGoal,
    required this.carbsGoal,
    required this.fatGoal,
    required this.fiberGoal,
  });

  final double calorieGoal;
  final double proteinGoal;
  final double carbsGoal;
  final double fatGoal;
  final double fiberGoal;

  Map<String, dynamic> toJson() {
    return {
      'calorieGoal': calorieGoal,
      'proteinGoal': proteinGoal,
      'carbsGoal': carbsGoal,
      'fatGoal': fatGoal,
      'fiberGoal': fiberGoal,
    };
  }
}
